File: go/1-basic-programming/2-xo.go
package main

import(
	"fmt"
)

func Xo(input string) bool {
	var countXo int
	
	for _, char := range input {
		if char == 'x' {
			countXo++
		}	
		if char == 'o' {
			countXo--
		}
	}

	if countXo == 0 {
		return true
	}
	return false
}

func main() {
	fmt.Println(Xo("xoxoxo")) // True
	fmt.Println(Xo("oxooxo")) // False
	fmt.Println(Xo("oxo")) // False
	fmt.Println(Xo("xxxooo")) // True
	fmt.Println(Xo("xoxooxxo")) // True
}

File: c++/1-basic/2-xo.cpp
#include <bits/stdc++.h>
using namespace std;

// complexity: O(n)
bool CountXO(string inputString) {
  int countX = 0, countY = 0;

  for (int i = 0; i < inputString.size(); i++) {
    if (inputString[i] == 'x') {
      countX += 1;
    } else if (inputString[i] == 'o') {
      countY += 1;
    }
  }
  
  if (countX == countY) {
    return true;
  }
  return false;
}

int main() {
  cout << CountXO("xoxoxo") << endl; // True
  cout << CountXO("oxooxo") << endl; // False
  cout << CountXO("oxo") << endl; // False
  cout << CountXO("xxxooo") << endl; // True
  cout << CountXO("xoxooxxo") << endl; // True
}